client/blocks/simple-site-rename-form/index.jsx

```diff
@@ -39,14 +44,44 @@ export class SimpleSiteRenameForm extends Component {
 		this.props.requestSiteRename( selectedSiteId, this.state.domainFieldValue, discard );
 	};
 
+	getDomainValidationMessage( domain ) {
+		const { translate } = this.props;
+
+		if ( isEmpty( domain ) ) {
+			return '';
+		}
+
+		if ( domain.match( /[^a-z0-9]/i ) ) {
+			return translate( 'Domain can only contain lowercase letters (a-z) and numbers.' );
+		}
+
+		if ( ! inRange( domain.length, SUBDOMAIN_LENGTH_MINIMUM, SUBDOMAIN_LENGTH_MAXIMUM ) ) {
+			return translate(
+				'Domain length should be between %(minimumLength)s and %(maximumLength)s characters.',
```

Comment:
Using the `maxLength` property on the `FormTextInputWithAffixes` component, you should be able to prevent them for inputting too many characters.

Reply:
Great suggestion, thanks! :)

Reply:
I gave this a whirl. It works as expected of course but we do lose the explicitness of showing the length message. Instead the person typing would just hit a limit and not be told why. Do you prefer either way in terms of UX?

Reply:
From a UX point of view, I'd argue we should always opt to take as much confusion out of a flow as possible. Based on that imo we should definitely go down the route of showing the additional error message/description.